File: ios/Bridge/Hybrid/TextRecognitionLanguage+toDomain.swift
extension TextRecognitionLanguage {
  func toDomainLanguage() -> DomainTextRecognitionLanguage {
    switch self {
    case .chinese:
      return .chinese
    case .devanagari:
      return .devanagari
    case .japanese:
      return .japanese
    case .korean:
      return .korean
    case .latin:
      return .latin
    }
  }
}

File: android/src/main/java/com/margelo/nitro/visioncameramlkit/TextRecognitionLanguage+toDomain.kt
package com.margelo.nitro.visioncameramlkit

internal fun TextRecognitionLanguage?.toDomainLanguage(): com.visioncameramlkit.domain.models.TextRecognitionLanguage =
  when (this) {
    TextRecognitionLanguage.CHINESE -> com.visioncameramlkit.domain.models.TextRecognitionLanguage.CHINESE
    TextRecognitionLanguage.DEVANAGARI -> com.visioncameramlkit.domain.models.TextRecognitionLanguage.DEVANAGARI
    TextRecognitionLanguage.JAPANESE -> com.visioncameramlkit.domain.models.TextRecognitionLanguage.JAPANESE
    TextRecognitionLanguage.KOREAN -> com.visioncameramlkit.domain.models.TextRecognitionLanguage.KOREAN
    TextRecognitionLanguage.LATIN,
    null,
    -> com.visioncameramlkit.domain.models.TextRecognitionLanguage.LATIN
  }
